### ISSUE TITLE
Validation metrics during training

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/AutoMONAI#31]
## Problem

Training currently has **no validation evaluation**. `run.py` trains on all of `imagesTr/` and only reports training loss/metrics. There is no way to detect overfitting, and early stopping + best model selection are based on training loss — which is unreliable.

## Current behaviour

- `train_one_epoch()` returns train loss and train metrics
- `best_model.pt` is saved when train loss improves
- Early stopping triggers when train loss stops improving
- No validation DataLoader exists in the training loop

### Validation loop

After each epoch (or every `--val_interval` epochs), run a no-grad forward pass on the validation set and compute val loss + val metrics.

```python
# In run.py training loop, after train_one_epoch():
if current_epoch % val_interval == 0 and val_loader is not None:
    val_result = validate(fabric, model, val_loader, loss_fn, metrics, num_classes)
```

Add `validate()` to `train.py`:
```python
def validate(fabric, model, val_loader, loss_fn, metrics, num_classes):
    """Run one validation pass. Returns dict with val_loss and val metrics."""
```

### What changes

| Component | Before | After |
|---|---|---|
| Best model selection | Train loss | Val loss (or val Dice) |
| Early stopping | Train loss stalls | Val loss stalls |
| W&B logging | `loss`, `dice` | `loss`, `dice`, `val/loss`, `val/dice` |
| Epoch log output | `Epoch 5/10 - Loss: 0.32 - dice: 0.71` | `Epoch 5/10 - Loss: 0.32 - dice: 0.71 - val_loss: 0.38 - val_dice: 0.65` |

### CLI flags
```
--val_interval    Run validation every N epochs (default: 1)
--best_metric     Metric to use for best model selection: val_loss | val_dice | train_loss (default: val_loss)
```

### Dependency on #27

This requires a train/val split to exist. Until #27 is implemented, a simple default hold-out split (80/20, seed 42) can be used as an interim solution so validation works out of the box.

### W&B logging
- `val/loss`, `val/dice`, `val/iou`, etc. logged per val interval
- `best_val_loss` and `best_val_dice` in run summary
